substrate-archive/src/database/listener.rs

```diff
@@ -122,13 +124,19 @@ where
 		let (tx, rx) = flume::bounded(1);
 		let pg_url = self.pg_url.clone();
 
-		let fut = async move {
+		// NOTE: this part is not included in the main future in order to prevent missing messages.
+		// Otherwise, it would be possible to spawn, immediately send a notification, which would be missed if we are not connected/listening yet.
+		let listener = task::block_on(async {
 			let mut listener = PgListener::connect(&pg_url).await?;
 			let channels = self.channels.iter().map(String::from).collect::<Vec<String>>();
 			listener.listen_all(channels.iter().map(|s| s.as_ref())).await?;
-			let mut conn = PgConnection::connect(&pg_url).await.unwrap();
+			Ok::<PgListener, ArchiveError>(listener)
+		})?;
```

Comment:
This looks good to me.

I'm just wondering; instead of blocking the current thread on `spawn`, would it make sense at some point to make `spawn` an `async fn` and await it in order to guarantee that it's ready before anything else happens?

Reply:
Yeah that would make sense, I think I had it that way in the past at some point in the past.

I transitioned to making the spawn blocking because it simplifies the startup, and by not making it async we can maintain a handle to the listener in the main `SystemInstance` object, which then allows us to clean it up on `Drop`. However it is very context specific and it probably would make more sense from an API standpoint to await this part and then use block_on if necessary outside of instantiation.